### ISSUE TITLE
Fetch error and autocompletion not working

### ISSUE THREAD

**Issue by tchereau** [microsoft/vscode-copilot-release#1043]
since 1 week, copilot doesn't fetch autocompletion anymore, there is just chat who is working
also while sometime toggle seem to work to re enable autocompletion, now it's just won't disable

- VS Code Version: 1.87.2
- OS Version: Sonoma 14.4 
- Logs:
```
2024-03-15 14:49:55.130 [info] [fetcher] Using Helix fetcher, Electron fetcher is not available.
2024-03-15 14:49:55.130 [info] [code-referencing] Public code references are enabled.
2024-03-15 15:18:35.571 [info] [fetcher] Using Helix fetcher, Electron fetcher is not available.
2024-03-15 15:18:35.571 [info] [code-referencing] Public code references are enabled.
2024-03-15 15:43:37.096 [info] [fetcher] Using Helix fetcher, Electron fetcher is not available.
2024-03-15 15:43:37.096 [info] [code-referencing] Public code references are enabled.
2024-03-15 15:47:40.942 [error] [default] github.copilot.toggleCopilot: Error: TypeError: Cannot assign to read only property '*' of object
    at Object.set (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:153:47905)
    at fM.updateEnabledConfig (/home/thomas/.vscode-server/extensions/github.copilot-1.173.0/extension/src/config.ts:75:31)
    at toggleCopilotEnablement (/home/thomas/.vscode-server/extensions/github.copilot-1.173.0/extension/src/config.ts:121:30)
    at fn (/home/thomas/.vscode-server/extensions/github.copilot-1.173.0/extension/src/extension.ts:73:15)
    at /home/thomas/.vscode-server/extensions/github.copilot-1.173.0/extension/src/telemetry.ts:24:26
    at u.h (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:150:184839)
    at u.g (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:150:183831)
    at u.executeCommand (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:150:183738)
    at Object.executeCommand (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:160:21654)
    at /home/thomas/.vscode-server/extensions/github.copilot-1.173.0/extension/src/statusBarPicker.ts:43:26
    at new Promise (<anonymous>)
    at OU.handleItemSelection (/home/thomas/.vscode-server/extensions/github.copilot-1.173.0/extension/src/statusBarPicker.ts:36:16)
    at p.value (/home/thomas/.vscode-server/extensions/github.copilot-1.173.0/extension/src/statusBarPicker.ts:29:18)
    at n.y (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:80:1902)
    at n.fire (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:80:2119)
    at s._fireDidAccept (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:150:157843)
    at n.$onDidAccept (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:150:156072)
    at s.S (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:147:5519)
    at s.Q (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:147:5285)
    at s.M (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:147:4375)
    at s.L (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:147:3454)
    at p.value (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:147:2241)
    at n.y (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:80:1902)
    at n.fire (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:80:2119)
    at r.fire (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:105:14091)
    at p.value (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:173:8050)
    at n.y (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:80:1902)
    at n.fire (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:80:2119)
    at r.fire (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:105:14091)
    at u.z (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:105:17156)
    at p.value (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:105:16561)
    at n.y (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:80:1902)
    at n.fire (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:80:2119)
    at p.acceptChunk (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:105:11865)
    at p.value (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:105:11152)
    at n.y (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:80:1902)
    at n.fire (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:80:2119)
    at n.z (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:170:19748)
2024-03-15 15:47:43.920 [error] [default] github.copilot.toggleCopilot: Error: TypeError: Cannot assign to read only property '*' of object
    at Object.set (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:153:47905)
    at fM.updateEnabledConfig (/home/thomas/.vscode-server/extensions/github.copilot-1.173.0/extension/src/config.ts:75:31)
    at toggleCopilotEnablement (/home/thomas/.vscode-server/extensions/github.copilot-1.173.0/extension/src/config.ts:121:30)
    at fn (/home/thomas/.vscode-server/extensions/github.copilot-1.173.0/extension/src/extension.ts:73:15)
    at /home/thomas/.vscode-server/extensions/github.copilot-1.173.0/extension/src/telemetry.ts:24:26
    at u.h (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:150:184839)
    at u.g (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:150:183831)
    at u.executeCommand (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:150:183738)
    at Object.executeCommand (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:160:21654)
    at /home/thomas/.vscode-server/extensions/github.copilot-1.173.0/extension/src/statusBarPicker.ts:43:26
    at new Promise (<anonymous>)
    at OU.handleItemSelection (/home/thomas/.vscode-server/extensions/github.copilot-1.173.0/extension/src/statusBarPicker.ts:36:16)
    at p.value (/home/thomas/.vscode-server/extensions/github.copilot-1.173.0/extension/src/statusBarPicker.ts:29:18)
    at n.y (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:80:1902)
    at n.fire (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:80:2119)
    at s._fireDidAccept (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:150:157843)
    at n.$onDidAccept (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:150:156072)
    at s.S (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:147:5519)
    at s.Q (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:147:5285)
    at s.M (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:147:4375)
    at s.L (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:147:3454)
    at p.value (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:147:2241)
    at n.y (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:80:1902)
    at n.fire (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:80:2119)
    at r.fire (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:105:14091)
    at p.value (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:173:8050)
    at n.y (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:80:1902)
    at n.fire (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:80:2119)
    at r.fire (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:105:14091)
    at u.z (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:105:17156)
    at p.value (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:105:16561)
    at n.y (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:80:1902)
    at n.fire (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:80:2119)
    at p.acceptChunk (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:105:11865)
    at p.value (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:105:11152)
    at n.y (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:80:1902)
    at n.fire (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:80:2119)
    at n.z (/home/thomas/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/out/vs/workbench/api/node/extensionHostProcess.js:170:19748)

```

**Comment by aeschli** [comment]
@digitarald Should be transferred to Github-Copilot

**Comment by devm33** [comment]
This should be fixed in 1.174.0

**Comment by aqlkzf** [comment]
## Copilot

- Version: 1.187.0
- Build: prod
- Editor: vscode/1.89.0

## Environment

- http_proxy: http://127.0.0.1:7890
- https_proxy: http://127.0.0.1:7890
- no_proxy: 127.0.0.1,login.microsoftonline.com,192.168.*,172.31.*,172.30.*,172.29.*,172.28.*,172.27.*,172.26.*,172.25.*,172.24.*,172.23.*,172.22.*,172.21.*,172.20.*,172.19.*,172.18.*,172.17.*,172.16.*,10.*,127.*,localhost,<local>
- SSL_CERT_FILE: n/a
- SSL_CERT_DIR: n/a
- OPENSSL_CONF: n/a

## Feature Flags

- Send Restricted Telemetry: enabled
- Chat: enabled

## Node setup

- Number of root certificates: 141
- Operating system: Linux
- Operating system version: 5.15.146.1-microsoft-standard-WSL2
- Operating system architecture: x64
- NODE_OPTIONS: n/a
- NODE_EXTRA_CA_CERTS: n/a
- NODE_TLS_REJECT_UNAUTHORIZED: n/a
- tls default min version: TLSv1.2
- tls default max version: TLSv1.3

## Network Configuration

- Proxy host: 127.0.0.1
- Proxy port: 7890
- Kerberos SPN: n/a
- Reject unauthorized: enabled
- Fetcher: HelixFetcher

## Reachability

- github.com: HTTP 200 - 
- copilot-proxy.githubusercontent.com: HTTP 200 - 
- api.githubcopilot.com: HTTP 200 - 
- default.exp-tas.com: HTTP 200 - 

## VS Code Configuration

- HTTP proxy: 
- HTTP proxy authentication: n/a
- Proxy Strict SSL: true
- Extension HTTP proxy support: override

## Extensions

- Is `win-ca` installed?: false
- Is `mac-ca` installed?: false

## Authentication

- GitHub username: aqlkzf
